File: Example/ExampleObjectiveC/PIOZoneViewController.m
//
//  PIOZoneViewController.m
//  ExampleObjectiveC
//
//  Created by Abdul Haseeb on 11/17/16.
//  Copyright © 2016 predict.io by ParkTAG GmbH. All rights reserved.
//

#import "PIOZoneViewController.h"
#import <MapKit/MapKit.h>

@interface PIOZoneViewController ()<MKMapViewDelegate>

@property (weak, nonatomic) IBOutlet MKMapView *mapView;

@end

@implementation PIOZoneViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    if (self.homeZone) {
        MKPointAnnotation *annotation = [[MKPointAnnotation alloc]init];
        annotation.coordinate = self.homeZone.center;
        annotation.title = @"Home Zone";
        [self.mapView addAnnotation:annotation];
        MKCircle *circle = [MKCircle circleWithCenterCoordinate:self.homeZone.center radius:self.homeZone.radius];
        [self.mapView addOverlay:circle];
    }
    
    if (self.workZone) {
        MKPointAnnotation *annotation = [[MKPointAnnotation alloc]init];
        annotation.coordinate = self.workZone.center;
        annotation.title = @"Work Zone";
        [self.mapView addAnnotation:annotation];
        MKCircle *circle = [MKCircle circleWithCenterCoordinate:self.workZone.center radius:self.workZone.radius];
        [self.mapView addOverlay:circle];
    }
}

- (void)viewDidAppear:(BOOL)animated {
    [self.mapView showAnnotations:self.mapView.annotations animated:YES];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (MKAnnotationView *)mapView:(MKMapView *)mapView
            viewForAnnotation:(id<MKAnnotation>)annotation {
    MKAnnotationView *annotationView = nil;
    if (!(annotation == mapView.userLocation)) {
        MKPinAnnotationView *pinAnnotationView = [[MKPinAnnotationView alloc] initWithAnnotation:annotation reuseIdentifier:@"zone"];
        pinAnnotationView.canShowCallout = YES;
        if ([annotation.title isEqualToString:@"Home Zone"]) {
            pinAnnotationView.pinTintColor = [UIColor orangeColor];
        } else {
            pinAnnotationView.pinTintColor = [UIColor purpleColor];
        }
        annotationView = pinAnnotationView;
    }
    return  annotationView;
}

- (MKOverlayRenderer *)mapView:(MKMapView *)mapView rendererForOverlay:(id<MKOverlay>)overlay {
    if ([overlay isKindOfClass:[MKCircle class]]){
        MKCircleRenderer *circleRenderer = [[MKCircleRenderer alloc] initWithCircle:overlay];
        circleRenderer.fillColor = [[UIColor greenColor] colorWithAlphaComponent:0.05];
        circleRenderer.strokeColor = [[UIColor greenColor] colorWithAlphaComponent:0.3];
        circleRenderer.lineWidth = 4;
        return circleRenderer;
    }
    return nil;
}

@end

File: ExampleSwift/ExampleSwift/PIOZoneViewController.swift
//
//  PIOZoneViewController.swift
//  ExampleSwift
//
//  Created by Zee on 16/11/2016.
//  Copyright © 2016 Abdul Haseeb. All rights reserved.
//

import UIKit
import MapKit
import PredictIO

class PIOZoneViewController: UIViewController {
    
    let homeZoneAnnotationTitle = "Home Zone"
    let workZoneAnnotationTitle = "Work Zone"
    
    var homeZone: PIOZone!
    var workZone: PIOZone!

    @IBOutlet weak var mapView: MKMapView!
    
    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
        if (homeZone != nil) {
            let annotation = MKPointAnnotation()
            annotation.coordinate = homeZone.center
            annotation.title = homeZoneAnnotationTitle
            mapView.addAnnotation(annotation)
            mapView.add(MKCircle(center: homeZone.center, radius: homeZone.radius))
        }
        
        if (workZone != nil) {
            let annotation = MKPointAnnotation()
            annotation.coordinate = workZone.center
            annotation.title = workZoneAnnotationTitle
            mapView.addAnnotation(annotation)
            mapView.add(MKCircle(center: workZone.center, radius: workZone.radius))
        }
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    func mapView(_ mapView: MKMapView, rendererFor overlay: MKOverlay) -> MKOverlayRenderer {
        if (overlay is MKCircle) {
            let circleRenderer = MKCircleRenderer.init(circle: overlay as! MKCircle)
            circleRenderer.fillColor = UIColor.green.withAlphaComponent(0.05)
            circleRenderer.strokeColor = UIColor.green.withAlphaComponent(0.3)
            circleRenderer.lineWidth = 4
            return circleRenderer
        }
        return MKPolylineRenderer()
    }
    
    func mapView(_ mapView: MKMapView, viewFor annotation: MKAnnotation) -> MKAnnotationView? {
        var annotationView: MKAnnotationView? = nil
        if !(annotation is MKUserLocation) {
            let pinAnnotationView = MKPinAnnotationView(annotation: annotation, reuseIdentifier: "zone")
            annotationView!.canShowCallout = true
            if (annotation.title! == homeZoneAnnotationTitle) {
                pinAnnotationView.pinTintColor = UIColor.orange
            } else {
                pinAnnotationView.pinTintColor = UIColor.purple
            }
            annotationView = pinAnnotationView
        }
        return annotationView!
    }
}
